src/main/groovy/org/boozallen/plugins/jte/console/TemplateLogger.groovy

```diff
@@ -42,23 +43,58 @@ public class TemplateLogger extends ConsoleNote<WorkflowRun> {
         return null
     }
 
-    static void print(String message, Boolean initiallyHidden = false, LogLevel logType = LogLevel.INFO) {
+    static void print(String message, Map config){
+        def argTypes = [
+                "initiallyHidden": Boolean,
+                "logType": LogLevel,
+                "trimLines": Boolean
+        ]
+
+        Boolean throwError = false
+        String errorMsg = "Improper TemplateLogger Configuration: \n"
+
+        def improperConfigNames = config.keySet() - argTypes.keySet()
+        if(improperConfigNames){
+            throwError = true
+            errorMsg +=  "The following keys are not identified: ${improperConfigNames} \n"
+            improperConfigNames.each{ config.remove(it) }
+        }
+        config.each{ key, value ->
+            Class expectedType = argTypes[key]
+            if(!(value.getClass() == expectedType)){
+                throwError = true
+                errorMsg += "key ${key} should be of type ${argTypes[key]} \n"
+            }
+        }
+
+        if (throwError){
```

Comment:
let's pull this exception out of the FileSystemWrapper - it was just the only thing that was using it so i had it as an inner class. 

it should really be it's own thing. 


on a more soap-boxy note - exception handling in JTE is kinda all over the place. 

I originally had started creating arbitrary exceptions purely for the sake of communication. 

TemplateBindingException for when it was an issue in the binding 
TemplateConfigException for when it was an issue w/ the config files

etc..

the different exceptions aren't really used for flow handling at all - just made it easier for me to look at the exception type and know where to start troubleshooting. 

we can consolidate this so everything uses JTEException and just make sure that the message is sufficient to find the root cause.

Reply:
agreed. I had intended JTEException to be the general JTE exception.  we can make the other jte Exceptions subclasses or replace.  Think it should be in another/new branch